pkg/ccl/backupccl/tenant_backup_nemesis_test.go

```diff
@@ -0,0 +1,505 @@
+						time.Sleep(time.Second)
+					}
+					if _, err := db.Exec("CANCEL JOB (SELECT job_id FROM [SHOW JOBS] WHERE job_type = 'IMPORT' AND status = 'paused')"); err != nil {
+						return err
+					}
+					if _, err := db.Exec("SHOW JOBS WHEN COMPLETE (SELECT job_id FROM [SHOW JOBS] WHERE job_type = 'IMPORT')"); err != nil {
+						return err
+					}
+					n.addTable(tableName)
+					return nil
+				},
+			},
+			{name: "IMPORT INTO",
+				impl: func(ctx context.Context, n *randomBackupNemesis, db *gosql.DB) error {
+					tableName, err := n.makeRandomBankTable("import_into")
+					if err != nil {
+						return err
+					}
+					n.addTable(tableName)
+					return nil
+				},
+			},
+			{name: "CREATE INDEX",
+				impl: func(ctx context.Context, n *randomBackupNemesis, db *gosql.DB) error {
+					tableName, err := n.makeRandomBankTable("create_index")
+					if err != nil {
+						return err
+					}
+					if _, err := db.Exec(fmt.Sprintf(`CREATE INDEX ON "%s"(n)`, tableName)); err != nil {
+						return err
+					}
+					n.addTable(tableName)
+					return nil
+				},
+			},
+			{name: "CREATE UNIQUE INDEX (will fail)",
+				impl: func(ctx context.Context, n *randomBackupNemesis, db *gosql.DB) error {
+					tableName, err := n.makeRandomBankTable("create_unique_index")
+					if err != nil {
+						return err
+					}
+					// Add a conflict that ensures the unique index creation will fail.
+					mkConflict := fmt.Sprintf(`UPDATE "%[1]s" SET n = 5 WHERE id IN (SELECT id FROM "%[1]s" ORDER BY random() LIMIT 5)`, tableName)
+					if _, err := db.Exec(mkConflict); err != nil {
+						return err
+					}
+
+					if _, err := db.Exec(fmt.Sprintf(`CREATE UNIQUE INDEX ON "%s"(n)`, tableName)); err == nil {
+						return errors.New("expected error but found none")
+					}
+					n.addTable(tableName)
+					return nil
+				},
+			},
+		},
+	}
+}
+
+func (r *randomBackupNemesis) Start(ctx context.Context) {
+	ctx, cancel := context.WithCancel(ctx)
+
+	r.cancel = cancel
+	r.grp = ctxgroup.WithContext(ctx)
+
+	r.grp.GoCtx(r.runNemesis)
+}
+
+func (r *randomBackupNemesis) Stop() {
+	r.cancel()
+	_ = r.grp.Wait()
+}
+
+func (r *randomBackupNemesis) RequireStart() chan struct{} {
+	notifyCh := make(chan nemesisNotification)
+	r.registerOneTimeListener(notifyCh)
+	newNemesis := <-notifyCh
+	return newNemesis.done
+}
+
+func (r *randomBackupNemesis) TablesToCheck() []string {
+	r.mu.Lock()
+	ret := append([]string(nil), r.mu.tablesToCheck...)
+	r.mu.Unlock()
+	return ret
+}
+
+func (r *randomBackupNemesis) addTable(name string) {
+	r.mu.Lock()
+	r.mu.tablesToCheck = append(r.mu.tablesToCheck, name)
+	r.mu.Unlock()
+}
+
+func (r *randomBackupNemesis) registerOneTimeListener(ch chan nemesisNotification) {
+	r.mu.Lock()
+	r.mu.oneTimeListeners = append(r.mu.oneTimeListeners, ch)
+	r.mu.Unlock()
+}
+
+func (r *randomBackupNemesis) notifyOneTimeListeners(done chan struct{}) {
+	r.mu.Lock()
+	for _, l := range r.mu.oneTimeListeners {
+		l <- nemesisNotification{
```

Comment:
why isn't the name field set in the nemesis notification?

Reply:
Didn't need it so I didn't send it :D.

Reply:
Eh, I'll just remove it. But an empty struct wouldn't work here because the point is to send the done channel